### PR TITLE
Create a set role color command and a ColorUtil class

### PR DESCRIPTION
**Pull request by a2937**

### CODE REVIEW

File: src/main/java/com/avairebot/commands/administration/SetRoleColorCommand.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        }
+
+        Role role = RoleUtil.getRoleFromMentionsOrName(context.message, args[0]);
+        if (role == null) {
+            return sendErrorMessage(context, context.i18n("noRoleFound", args[0]));
+        }
+
+        if (args[1].equalsIgnoreCase("random")) {
+            Color color = RandomUtil.getRandomColor();
+            RoleImpl actualRole = (RoleImpl) role;
```
Comment (Senither) [inline src/main/java/com/avairebot/commands/administration/SetRoleColorCommand.java]: You don't need to cast the role instance here, you can just call:
```java
role.getManager().setColor(someColor).queue();
```

File: src/main/java/com/avairebot/commands/administration/SetRoleColorCommand.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    @Override
+    public List<String> getTriggers() {
+        return Collections.singletonList("setrolecolor");
+    }
+
+    @Override
+    public List<String> getMiddleware() {
+        return Arrays.asList(
+            "require:user,general.administrator",
```
[the remark this answers is not in the document]
Comment (a2937) [reply src/main/java/com/avairebot/commands/administration/SetRoleColorCommand.java]: This has been replaced.

File: src/main/java/com/avairebot/commands/administration/SetRoleColorCommand.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            Color color = ColorUtil.getColorFromString(args[1].toUpperCase());
+            if (color != null) {
+                RoleImpl actualRole = (RoleImpl) role;
+                actualRole.getManager().setColor(color).queue();
+            } else {
+                return sendErrorMessage(context, context.i18n("invalidColor", args[1]));
+            }
+        }
+
+        return true;
```
[the remark this answers is not in the document]
Comment (a2937) [reply src/main/java/com/avairebot/commands/administration/SetRoleColorCommand.java]: I decided to go with the format you provided, and rely on what the user has inserted.

File: src/main/java/com/avairebot/commands/administration/SetRoleColorCommand.java (hunk 1 above)
Comment (a2937) [reply src/main/java/com/avairebot/commands/administration/SetRoleColorCommand.java]: I removed the cast and the RoleImpl instance.

**Pull request merged**